### PR TITLE
Force the linter as function context

### LINKED ISSUE #1048: Plugin lint function does not get the good "this"
Since version 2.0 (nodejs 5.1.0), it seems that we don't get the good 'this' object in the lint function 

```
var TypescriptCompiler = function (config) {
    this.config = "config";

    console.log('init',this)
};

TypescriptCompiler.prototype.brunchPlugin = true;
TypescriptCompiler.prototype.type = 'javascript';
TypescriptCompiler.prototype.extension = "ts";

TypescriptCompiler.prototype.lint = function (data, path, callback) {
  console.log("lint ",this.config)
  return callback(null,null)
}
```

Display : 
```
λ brunch build
init { config: 'config' }
lint  undefined
```

'this' is still good in the compile function.

### PR DESCRIPTION
**Pull request by ghostd**
Fix #1048

### CODE REVIEW

File: lib/fs_utils/pipeline.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const lint = (data, path, linters) => {
   const reduceLinter = (previousPromise, linter) => {
     debug(`Linting '${path}' with '${linter.constructor.name}'`);
     return previousPromise.then(() => {
-      return promisify(linter.lint)(data, path);
+      return promisify(linter.lint).apply(linter, [data, path]);
```
Comment (es128) [inline lib/fs_utils/pipeline.js]: why not `.call(linter, data, path)`?

**Comment by ghostd** [comment]
You're right. I'm going to change that.

**Comment by ghostd** [comment]
@es128 done

**Comment by paulmillr** [comment]
Perfect. Thanks dude.

**Pull request merged**